File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int l = 0;

        for (auto r = 0; r < nums.size(); r++) {
            if (nums[l] != nums[r]) {
                l++;
                swap(nums[l], nums[r]);
            }
        }
        return l + 1;
    }
};

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.go
func removeDuplicates(nums []int) int {
    s := 0

    for f := 0; f < len(nums); f++ {
        if nums[f] != nums[s] {
            s++
            nums[s], nums[f] = nums[f], nums[s]
        }
    }

    return s + 1
}